service/awsclusterconfig/v1/resource_set.go

```diff
@@ -66,6 +88,7 @@ func NewResourceSet(config ResourceSetConfig) (*framework.ResourceSet, error) {
 
 	resources := []framework.Resource{
 		awsConfigResource,
+		encryptionKeyResource,
```

Comment:
IMO the encryption key stuff should be created before the bigger cluster creation part because the encryption key thing is faster and depending on the implementation of the provider specific cluster resources we get into a "dead lock" where the first reconciliation fails because the encryption key is not there where the cluster resources require them.

Reply:
I understand your point here, but is this something that should be documented on OperatorKit: https://github.com/giantswarm/operatorkit/blob/master/framework/resource_set.go#L30-L32 or would it be possible to actually run resources concurrently (I understand that this requires changes to OperatorKit currently, but still asking)?

Reply:
In the future we can think about scheduling of resource execution and how to define dependencies and concurrency. I thought about something like a middleware like approach where you define a graph or something, but this should be a separate discussion. 

IMO it is stated quite clearly that the resource execution is sequential. When we have to improve the docs because this might not be clear enough then lets improve this. On a quick scan I just found this: https://github.com/giantswarm/operatorkit/blob/ab7825b0d283f56220137f4d5c9c7e9b35ed59ac/framework/resource.go#L7-L8.